File: abc/abc335/b.java
// https://atcoder.jp/contests/abc335/submissions/49174417

import java.util.Scanner;

class Main {
  public static void main(String args[]) {
    Scanner stdIn = new Scanner(System.in);

    int n = stdIn.nextInt();
    for(int i = 0; i <= n; i++) {
      for(int j = 0; j <= n; j++) {
        for(int k = 0; k <= n; k++) {
          if(i + j + k <= n) {
            System.out.printf("%d %d %d\n", i, j, k);
          }
        }
      }
    }
  }
}

File: abc/abc335/b.py
# https://atcoder.jp/contests/abc335/submissions/49174390

n = int(input())

ans = set()
for i in range(n + 1):
  for j in range(n + 1):
    for k in range(n + 1):
      if i + j + k <= n:
        print(i, j, k)
